src/ops/fuelup_self.rs

```diff
@@ -0,0 +1,20 @@
+use anyhow::{bail, Result};
+
+use crate::{
+    download::{component, download_file_and_unpack, unpack_bins, DownloadCfg},
+    path::fuelup_bin_dir,
+};
+
+pub fn self_update() -> Result<()> {
+    let download_cfg = DownloadCfg::new(component::FUELUP, None)?;
+    let fuelup_bin_dir = fuelup_bin_dir();
+    if download_file_and_unpack(&download_cfg, &fuelup_bin_dir).is_err() {
+        bail!("{} {}", &download_cfg.name, &download_cfg.version)
```

Comment:
It looks like this (and the following `bail!`) might just error with the name and version but no details. Should we give more context? E.g.

```rust
if let Err(e) = download_file_and_unpack(&download_cfg, &fuelup_bin_dir) {
    bail!("failed to download and unpack {}-{}: {}", download_cfg.name, download_cfg.version, e);
```

Reply:
Thanks for the catch! Missed this but the intention was to bubble up the errors to `commands/fuelup.rs` since there were already error messages within `download_file_and_unpack` and also in `unpack`. Commit [169d24c](https://github.com/FuelLabs/fuelup/pull/73/commits/169d24c3a10b55b26d3d6a013c792845eaf8782a) should fix this